IdentityCore/src/validation/MSIDAuthorityFactory.m

```diff
@@ -70,4 +71,17 @@ + (MSIDAuthority *)authorityFromUrl:(NSURL *)url
     return nil;
 }
 
++ (MSIDAuthority *)authorityFromJSONDictionary:(NSDictionary *)json
+                                         error:(NSError **)error
+{
+    NSString *authorityType = [json msidStringObjectForKey:MSID_AUTHORITY_TYPE_JSON_KEY];
+    
+    // If there is no authority type, we assume it's AAD authority.
+    if (!authorityType) authorityType = [MSIDAADAuthority authorityType];
+    
+    MSIDAuthority *authority = (MSIDAuthority *)[MSIDJsonSerializableFactory createFromJSONDictionary:json classTypeKey:authorityType error:error];
```

Comment:
this should use classValueKey method

Reply:
Removed this logic from the factory. We will preprocess json on broker side to make sure authority url & type is always in json payload.